pkg/transport/client.go

```diff
@@ -0,0 +1,13 @@
+package transport
+
+import (
+	"net/http"
+)
+
+// Client is a holder of an http.Client.  In future it will have some high level logic
+// attached to it (PostProtobuf, PostJson, retries, metrics, etc).  The underlying Client
+// is exposed so that things that require a real http.Client (such as Cloudwatch) can
+// still utilize the TransportPool.
```

Comment:
This is a very leaky abstraction. Everywhere `transport.Client` is used it's to call `Do` on the underlying `http.Client`. I'd suggest you might as well embrace this and embed `http.Client` here, but on its own that doesn't solve the cloudwatch case.

Alternatives:

- for this first cut I think you can remove `transport.Client`(using `http.Client` instead) and in future introduce what you need to support other features / transports
- implement `Do` on `transport.Client` to call the underlying Client, which at least eliminates some stutter

Reply:
Since generally all (most) of the `x.Client.Do`s and surrounding code will be removed anyway, I'll switch to just storing `pool.Get(...).Client` for now.  It will also simplify the diff a bit, because the actual `client.Do` won't change at all then.